### PR TITLE
Add inner_iterations_count result option to LogisticRegression

### PR DESCRIPTION
**Pull request by avolkov-intel**
# Description
To access the number of iterations done by cg-solver on the python side we need to add a new result_option to oneDAL algorithm

### CODE REVIEW

File: cpp/oneapi/dal/algo/logistic_regression/backend/gpu/train_kernel_dense_batch_dpc.cpp [hunk 1]
```diff
@@ -88,6 +88,8 @@ static train_result<Task> call_dal_kernel(const context_gpu& ctx,
 
     auto [train_event, iter_num] = opt_impl->minimize(queue, loss_func, x_suf, { fill_event });
 
+    train_event.wait_and_throw();
```
Comment (ethanglaser) [inline cpp/oneapi/dal/algo/logistic_regression/backend/gpu/train_kernel_dense_batch_dpc.cpp:91]: is it necessary?

Comment (Alexandr-Solovev) [inline cpp/oneapi/dal/algo/logistic_regression/backend/gpu/train_kernel_dense_batch_dpc.cpp:91]: Please, add TODO for investigate the reason of sporadic failures(probably it affects other algorithms as well)

**Review by Alexandr-Solovev — APPROVED** [review]
LGTM, may be it makes sense to add check for new result_option. As we do for descriptors `is_valid_method_v`, but can be added with adding new method

**Comment by avolkov-intel** [comment]
http://intel-ci.intel.com/eedbbc6a-08f3-f1b5-86db-a4bf010d0e2e

**Pull request merged**